evaluate/report/csv_test.go

```diff
@@ -154,3 +177,602 @@ func TestWriteEvaluationRecord(t *testing.T) {
+					metrics.AssessmentKeyResponseNoExcess:                   49,
+					metrics.AssessmentKeyResponseWithCode:                   50,
+				},
+			},
+		},
+	})
+	validate(t, &testCase{
+		Name: "Multiple assessments",
+
+		Before: func(resultPath string) {
+			fileContent := bytesutil.StringTrimIndentations(`
+					model-id,model-name,cost,language,repository,task,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
+					openrouter/anthropic/claude-1.2,Claude 1.2,0.0001,golang,golang/light,write-tests,982,750,18,70179,720571,71195,115,49,50
+					openrouter/anthropic/claude-1.2,Claude 1.2,0.0002,golang,golang/plain,write-tests,37,20,2,441,11042,523,5,5,5
+				`)
+			require.NoError(t, os.WriteFile(filepath.Join(resultPath, "evaluation.csv"), []byte(fileContent), 0644))
+		},
+
+		ExpectedEvaluationRecords: EvaluationRecords{
+			&EvaluationRecord{
+				ModelID:    "openrouter/anthropic/claude-1.2",
+				ModelName:  "Claude 1.2",
+				ModelCost:  0.0001,
+				LanguageID: "golang",
+				Assessments: metrics.Assessments{
+					metrics.AssessmentKeyCoverage:                           750,
+					metrics.AssessmentKeyFilesExecuted:                      18,
+					metrics.AssessmentKeyGenerateTestsForFileCharacterCount: 70179,
+					metrics.AssessmentKeyProcessingTime:                     720571,
+					metrics.AssessmentKeyResponseCharacterCount:             71195,
+					metrics.AssessmentKeyResponseNoError:                    115,
+					metrics.AssessmentKeyResponseNoExcess:                   49,
+					metrics.AssessmentKeyResponseWithCode:                   50,
+				},
+			},
+			&EvaluationRecord{
+				ModelID:    "openrouter/anthropic/claude-1.2",
+				ModelName:  "Claude 1.2",
+				ModelCost:  0.0002,
+				LanguageID: "golang",
+				Assessments: metrics.Assessments{
+					metrics.AssessmentKeyCoverage:                           20,
+					metrics.AssessmentKeyFilesExecuted:                      2,
+					metrics.AssessmentKeyGenerateTestsForFileCharacterCount: 441,
+					metrics.AssessmentKeyProcessingTime:                     11042,
+					metrics.AssessmentKeyResponseCharacterCount:             523,
+					metrics.AssessmentKeyResponseNoError:                    5,
+					metrics.AssessmentKeyResponseNoExcess:                   5,
+					metrics.AssessmentKeyResponseWithCode:                   5,
+				},
+			},
+		},
+	})
+}
+
+func TestEvaluationRecordsGroupByModel(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		EvaluationRecords EvaluationRecords
+
+		ExpectedEvaluationRecords map[string]*EvaluationRecord
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			actualEvaluationRecords := tc.EvaluationRecords.GroupByModel()
+
+			expected, actual := maps.Clone(tc.ExpectedEvaluationRecords), maps.Clone(actualEvaluationRecords)
```

Comment:
Why do we need to clone?

Reply:
It looks like we still need the clone otherwise the scoring is not correct. That's because in the `WriteCSVs` we are grouping the records in two different ways and we are using the original evaluation record pointers, so we need to clone them.

Reply:
If we need something special in the tests to make it work, then we should solve that in the actual code. Cause that means we could forget to do that in the real code :see_no_evil:

Reply:
Overlooked the comment, I thought you were talking about the `Clone` function for `EvaluationRecords`.

Reply:
Removed the `maps.Clone` since it is not needed